execution/execution.go

```diff
@@ -211,8 +215,7 @@ func newOperator(expr parser.Expr, storage *engstore.SelectorPool, opts *query.O
 		return newOperator(e.Expr, storage, opts, hints)
 
 	case *parser.StringLiteral:
-		// TODO(saswatamcode): This requires separate model with strings.
-		return nil, errors.Wrapf(parse.ErrNotImplemented, "got: %s", e)
+		return nil, errors.Newf("string literals don't support operators: %s", e)
```

Comment:
Is it possible to hit this code path? Maybe we can remove the `*parser.StringLiteral` case completely now that we handle strings in the engine itself.